designer/client/context/__tests__/FeatureFlagContext.jest.tsx

```diff
@@ -0,0 +1,145 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import { Router } from "react-router-dom";
+import { act } from "react-dom/test-utils";
+import { FeatureFlagProvider } from "../FeatureFlagContext";
+import { createMemoryHistory } from "history";
+import sinon from "sinon";
+import { FeatureToggleApi } from "../../api/toggleApi";
+import FeatureToggle from "../../FeatureToggle";
+import { useFeatures } from "../../hooks/featureToggling";
+const history = createMemoryHistory();
+history.push("");
+
+const customRender = (ui, { providerProps, ...renderOptions }) => {
+  const rendered = render(
+    <Router history={history}>
+      <FeatureFlagProvider value={providerProps}>{ui}</FeatureFlagProvider>
+    </Router>,
+    renderOptions
+  );
+  return {
+    ...rendered,
+    rerender: (ui, options) =>
+      customRender(ui, { container: rendered.container, ...options }),
+  };
+};
+
+describe("FeatureFlagContext", () => {
+  beforeEach(() => {
+    sinon.restore();
+  });
+
+  const WrappingComponent = ({ children }) => {
+    return <FeatureFlagProvider>{children}</FeatureFlagProvider>;
+  };
+
+  test("should show element if feature is set", async () => {
+    sinon
+      .stub(FeatureToggleApi.prototype, "fetch")
+      .callsFake(async function () {
+        return { featureEditPageDuplicateButton: true };
+      });
+
+    const { findAllByText } = render(
+      <WrappingComponent>
+        <FeatureToggle feature="featureEditPageDuplicateButton">
+          <button>Johnny Five Is Alive!</button>
+        </FeatureToggle>
+      </WrappingComponent>
+    );
+    expect(await findAllByText("Johnny Five Is Alive!")).toBeTruthy();
+  });
+
+  test("should not show element if feature is not set", async () => {
+    const stub = sinon
+      .stub(FeatureToggleApi.prototype, "fetch")
+      .callsFake(async function () {
+        return { featureEditPageDuplicateButton: false };
+      });
```

Comment:
this test is only testing of the values provided by the context is getting propagated to the component - so mocking this call is probably OTT?  You can just wrap your component with
```suggestion 
<Context.Provider value={stubbedValues}>
{children}
</Context.Provider>
```